### ISSUE TITLE
Include modular packages with broken deps in the output

### ISSUE THREAD

**Issue by lubomir** [fedora-modularity/fus#48]
If the package is listed explicitly as input, it will be included even if it has unresolvable dependencies.

If a modular package has broken deps, it will not be listed. However since it's basically also an input solvable, it would be nice to list it in the output. The NVRA is known, so it should be possible.

**Comment by r4f4** [comment]
It *will* be listed. The problem is that in the tests, a `g_warning` generates an abort so `fus` doesn't get to the loop for printing solvables in the pile.
If we move the line
```
if (solv_failed)
   g_warning ("Can't resolve all solvables");
````
 to after the package output loop, it would be printed. Here is what the output would be (and is currently in the main fus):
```
$: ./fus-check --repo repo,repo,$PWD/tests/module-broken/packages.repo --repo yaml,modular,$PWD/tests/module-broken/modules.yaml --platform f29 "module(m:master)"
Problem 1 / 1:
  - conflicting requests
  - nothing provides bar needed by foo-1.0-1.x86_64

(fus-check:18424): fus-WARNING **: 22:31:41.455: Can't resolve all solvables
module:m:master:20180904161631:cafebabe.x86_64@yaml
```

**Comment by lubomir** [comment]
But your output only shows the module itself, not the package `foo` that is in the module.

**Comment by r4f4** [comment]
Oh ok, I misunderstood the problem.

**Comment by r4f4** [comment]
I fixed the tests to not abort on `g_warning` in 33fe53e2623198d5a768810fb2a8552cfa4d11b4. That partly fixes the problem by outputting the solvables in the input even when they have broken deps.

**Comment by lubomir** [comment]
It's definitely an improvement. However what I would like to see is in the `/fail/module/broken` test case the output to also include the `foo` package.

It would also affect the `/fail/moddep/broken` case. The current behaviour is a bit weird: you ask for module C, D and E, and get in output, but not any of their packages, so they are basically empty.

**Comment by r4f4** [comment]
The problem here is that even if you ignore the problems in `solver_solve`, `solver_create_transaction` does not give you the broken packages as candidates to install.
```
fus-DEBUG: 11:28:42.464: Searching combinations for module:E:master:2018:deadbeef.x86_64
fus-DEBUG: 11:28:42.464:   Transaction 1 / 1:
fus-DEBUG: 11:28:42.464:     - module:E:master:2018:deadbeef.x86_64
fus-DEBUG: 11:28:42.464:     - module:platform:f29:0:00000000.x86_64
fus-DEBUG: 11:28:42.464:     Installing E-1.0-1.x86_64:
Problem 1 / 1:
  - package E-1.0-1.x86_64 requires C, but none of the providers can be installed
  - package C-1.0-1.x86_64 requires module:C:master:2018:deadbeef.x86_64, but none of the providers can be installed
  - conflicting requests
  - nothing provides (module(D) with module(D:stable)) needed by module:C:master:2018:deadbeef.x86_64
fus-DEBUG: 11:28:42.464: Instral trans: 2
fus-DEBUG: 11:28:42.464:        - module:E:master:2018:deadbeef.x86_64
fus-DEBUG: 11:28:42.464:        - module:platform:f29:0:00000000.x86_64
```
I don't know how to ask libsolv to return all packages even if they have broken deps.

**Comment by lubomir** [comment]
I understand the candidates are not available and I agree that is not a simple thing to solve and won't happen any time soon. What I meant is the list of RPM artifacts in the modulemd. That is available already even before libsolv starts solving anything.

**Comment by r4f4** [comment]
@lubomir what do you think of this last commit?

**Comment by lubomir** [comment]
That looks very good! I'll test it on real data and report back status.

**Comment by r4f4** [comment]
@lubomir how did it go with real data?

**Comment by lubomir** [comment]
It works exactly how I imagined. If there is a package in a module, but can't be installed, it still gets into the compose. :+1: from to to get this patch into master.

**Comment by r4f4** [comment]
Fixed by #52
